### ISSUE TITLE
Allow self-signed SSL certificate for HTTPS HTML source

### ISSUE THREAD

**Issue by clsty** [ImranR98/Obtainium#1258]
I have a Linux server running on a local network, and its domain can only be accessed within the local network.

The corresponding HTTPS certificate is signed by a self-built rootCA called mkcert.

After adding the mkcert rootCA to the trusted list of Android phone, the Obtainium App still cannot connect to the HTML source which URL protocol is HTTPS and is provided by this Linux server and reports the following error:
```plain
HandshakeException: Handshake error in client (OS Error: CERTIFICATE_VERIFY_FAILED: unable to get local issuer certificate(handshake.cc:393))
```

In conclusion, I hope the rootCA could be added to Obtainium App in its settings, or at least allow skip SSL check for specific HTML source.

**Comment by ImranR98** [comment]
Blocked by https://github.com/dart-lang/sdk/issues/50435

**Comment by Glazami** [comment]
```
HandshakeException: Handshake error in client (OS Error: 
 SSLV3_ALERT_HANDSHAKE_FAILURE(tls_record.cc:583)
 HANDSHAKE_FAILURE_ON_CLIENT_HELLO(handshake.cc:657)) 
```

**Comment by fayezk13** [comment]
@ImranR98 there is a solution proposed here by the blocking issue: https://github.com/dart-lang/sdk/issues/50435#issuecomment-2455385431
could that be tested. Thanks.
